core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -190,7 +192,8 @@ private[kafka] object ConsumerStage {
       if (offsets.isEmpty)
         Future.successful(Done)
       else {
-        stages.head._2.commit(this)
+        implicit val ec = akka.dispatch.ExecutionContexts.global()
+        Future.sequence(stages.map(_._2.commit(this))).map(_ => Done)
```

Comment:
`ExecutionContext.global` should not be used, we should stay in Akka EC land. Hmm, I think I did this in one of the earlier implementations, but it might have been lost in some refactoring. I'll try to find it.